app/src/main/java/cmpe277/project/skibuddy/SignInActivity.java

```diff
@@ -0,0 +1,258 @@
+                e.printStackTrace();
+            }
+        } else {
+            showProgressDialog();
+            opr.setResultCallback(new ResultCallback<GoogleSignInResult>() {
+                @Override
+                public void onResult(GoogleSignInResult googleSignInResult) {
+                    hideProgressDialog();
+                    try {
+                        handleSignInResult(googleSignInResult);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    } catch (NoUserIdException e) {
+                        e.printStackTrace();
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == RC_SIGN_IN) {
+            GoogleSignInResult result = Auth.GoogleSignInApi.getSignInResultFromIntent(data);
+
+            try {
+                handleSignInResult(result);
+            } catch (IOException e) {
+                e.printStackTrace();
+            } catch (JSONException e) {
+                e.printStackTrace();
+            } catch (NoUserIdException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Exception thrown when no user ID could be retrieved.
+     */
+    private static class NoUserIdException extends Exception {
+    }
+
+    private void handleSignInResult(GoogleSignInResult result) throws IOException, JSONException, NoUserIdException {
+        Log.d(TAG, "handleSignInResult:" + result.isSuccess());
+        if (result.isSuccess()) {
+            Intent userProfileIntent = new Intent(SignInActivity.this, UserProfileActivity.class);
+            ArrayList<String> arrStr = new ArrayList<String>();
+
+            if (mGoogleApiClient.hasConnectedApi(Plus.API)) {
+                Person currentPerson = Plus.PeopleApi.getCurrentPerson(mGoogleApiClient);
+                final String name = currentPerson.getDisplayName();
+                final String taglineStr = currentPerson.getTagline();
+                final String uriStr = currentPerson.getImage().getUrl();
+
+                //TODO: to be deprecated
+                arrStr.add(name);       //index 0
+                arrStr.add(taglineStr); //index 1
+                arrStr.add(uriStr);     //index 2
+
+
+                userProfileIntent.putExtra("UserProfileActivity", arrStr);
+//              SignInActivity.this.startActivity(userProfileIntent);
+                updateUI(true);
+                //TODO: end of to be deprecated
+
+                //instead of user profile, load dashboard after sign in
+                final Server s = new ServerSingleton().getServerInstance(this);
+                final String UID = "qwes";
```

Comment:
Could you change this to "G+" + Google+ profile ID number, it should be unique for each user. I just put "qwes" as a temporary thing for testing.

Thanks!